src/components/Plate/Plate.tsx

```diff
@@ -1,67 +1,101 @@
+import { AnnotationBar } from "./AnnotationBar";
+import { useState } from "react";
 
 const PlateSelectionSchema = z.object({
   wells: z.array(z.number()),
   className: z.string().optional(),
 });
 export type PlateSelection = z.infer<typeof PlateSelectionSchema>;
 
-export interface PlateProps {
+export interface WellAnnotation<T extends Record<string, unknown>> {
+  id: string;
+  wells: number[];
+  label: string;
+  className?: string;
+  metadata?: T;
+}
+export interface RowAnnotation<T extends Record<string, unknown>> {
+  id: string;
+  rows: number[];
+  label: string;
+  className?: string;
+  metadata?: T;
+}
+export interface ColAnnotation<T extends Record<string, unknown>> {
+  id: string;
+  cols: number[];
+  label: string;
+  className?: string;
+  metadata?: T;
+}
+
+export interface PlateProps<
+  RowMetaT extends Record<string, unknown>,
+  ColMetaT extends Record<string, unknown>,
+  WellMetaT extends Record<string, unknown>,
+> {
   wells: 24 | 96 | 48 | 384;
-  rowAnnotations?: RowAnnotation[];
+  rowAnnotations?: RowAnnotation<RowMetaT>[];
+  colAnnotations?: ColAnnotation<ColMetaT>[];
+  wellAnnotations?: WellAnnotation<WellMetaT>[];
   selection: PlateSelection | null;
   setSelection: (selection: PlateSelection | null) => void;
   className?: string;
   selectionTolerance?: number;
 }
 
-export const Plate = ({
+export const Plate = <
+  RowMetaT extends Record<string, unknown>,
+  ColMetaT extends Record<string, unknown>,
+  WellMetaT extends Record<string, unknown>,
+>({
   wells,
   className,
   rowAnnotations,
+  colAnnotations,
+  wellAnnotations,
   selection,
   setSelection,
   selectionTolerance = 20,
-}: PlateProps) => {
+}: PlateProps<RowMetaT, ColMetaT, WellMetaT>) => {
   const { rows, cols } = wellsToRowsCols(wells);
+  const [activeRowAnnotation, setActiveRowAnnotation] =
+    useState<RowAnnotation<RowMetaT> | null>(null);
+  const [activeColAnnotation, setActiveColAnnotation] =
+    useState<ColAnnotation<ColMetaT> | null>(null);
+  const [activeWellAnnotation, setActiveWellAnnotation] =
+    useState<WellAnnotation<WellMetaT> | null>(null);
+  console.log(
+    activeRowAnnotation?.id,
+    activeColAnnotation?.id,
+    activeWellAnnotation?.id,
+  );
   const rowLabels: string[] = Array.from({ length: rows }, (_, i) =>
     (i + 1).toString(),
   );
   const colLabels: string[] = Array.from({ length: cols }, (_, i) =>
     String.fromCharCode(65 + i),
   );
 
-  // Num cols starts with one to support the row label column
-  var numCols = 1;
-
-  var numRowAnnotationCols = 0;
-  // If there are row annotations, add a column to display them.
-  if (rowAnnotations && rowAnnotations.length > 0) {
-    // The number of columns that the row annotations will span
-    numRowAnnotationCols = Math.ceil(rowAnnotations.length / 8);
-
-    // Each column can support up to 8 row annotations
-    numCols += numRowAnnotationCols;
-  }
-
   // Add the number of columns for the wells
+  let gridClass: string;
```

Comment:
Nice work getting the sub grid sorted so that the actual plate grid layout could be static classess. I knew i was missing something!